cylc/flow/xtriggers/workflow_state.py

```diff
@@ -14,18 +14,127 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import sqlite3
-from typing import Dict, Optional, Tuple
+from typing import Dict, Optional, Tuple, Any
+import asyncio
+from inspect import signature
 
-from metomi.isodatetime.parsers import TimePointParser
-
-from cylc.flow.cycling.util import add_offset
-from cylc.flow.dbstatecheck import CylcWorkflowDBChecker
-from cylc.flow.pathutil import get_cylc_run_dir
-from cylc.flow.workflow_files import infer_latest_run_from_id
+from cylc.flow.scripts.workflow_state import WorkflowPoller
+from cylc.flow.id import tokenise
+from cylc.flow.exceptions import WorkflowConfigError
+from cylc.flow.task_state import TASK_STATUS_SUCCEEDED
 
 
 def workflow_state(
+    workflow_task_id: str,
+    offset: Optional[str] = None,
+    flow_num: Optional[int] = None,
+    is_output: bool = False,
+    is_message: bool = False,
+    alt_cylc_run_dir: Optional[str] = None,
+) -> Tuple[bool, Dict[str, Any]]:
+    """Connect to a workflow DB and check a task status or output.
+
+    If the status or output has been achieved, return {True, result}.
+
+    Arg:
+        workflow_task_id:
```

Comment:
Can you describe `is_message` in the docstring?

Reply:
And `is_output`

Reply:
Done. And I changed is_output to is_trigger (everywhere) in light of us recently settling on "output" = "trigger: message".